doc/neps/nep-0010-new-iterator-ufunc.rst

```diff
@@ -575,12 +575,16 @@ ndim, and niter will produce slightly different layouts.
             intp shape;
             /* The current coordinate along this axis */
             intp coord;
-            /* The operand and index strides for this axis
+            /* The operand and index strides for this axis */
             intp stride[niter];
-            {intp indexstride;} #if (flags&FLAGS_HASINDEX);
```

Comment:
I'm not sure what the intent here was, I assume what I replaced it with but this looked pretty weird.

Reply:
I think you got it right, but it is a bit odd to use the same name for two different structures.

Reply:
I can't work out what your comment is referring to @charris. What name and which structures?

Reply:
Looking in the code, there doesn't seem to be anything like this, so I expect it has marginal relevance to how the implementation ended up. The closest thing I could find is
```
/* Internal-only AXISDATA MEMBER ACCESS. */
struct NpyIter_AD {
    npy_intp shape, index;
    npy_intp ad_flexdata;
};
```
Where `ad_flexdata` seems to be either a stride, or a pointer stored as an integer. It's probably going to cause trouble eventually.

Reply:
@eric-wieser It looked like `struct axis_data` could contain different entries depending on a flag, I think `NpyIter_AD` has the same dual use of `ad_flexdata`.

Reply:
Thanks for checking. I think no need to look into it more. Just to document why I changed it: 
- it caused a Sphinx warning
- it caused code highlighting for the whole code block to go missing